www/js/init.js

```diff
@@ -33,8 +33,176 @@
+        setTimeout(function () {
+            var kiwixComm = document.getElementById('kiwixComm');
+            // The only browser which does not support .remove() is IE11, but it will never run this code
+            if (kiwixComm) kiwixComm.remove();
+        }, 3000);
+    }
+})();
+
+// Since contentInjectionMode can be overriden when returning from remote PWA to extension (for example), we have to prevent an infinite loop
+// with code that warns the user to turn off the App Cache bypass in jQuery mode. Note that to turn OFF the bypass, we have to set the VALUE to true
+params.appCache = params.contentInjectionMode === 'jquery' ? true : params.appCache;
+
+/**
+ * Set the State and UI settings associated with parameters defined above
+ */
+document.getElementById('hideActiveContentWarningCheck').checked = params.hideActiveContentWarning;
+document.getElementById('disableDragAndDropCheck').checked = params.disableDragAndDrop;
+document.getElementById('showUIAnimationsCheck').checked = params.showUIAnimations;
+document.getElementById('titleSearchRange').value = params.maxSearchResultsSize;
+document.getElementById('titleSearchRangeVal').textContent = params.maxSearchResultsSize;
+document.getElementById('appThemeSelect').value = params.appTheme;
+document.getElementById('useHomeKeyToFocusSearchBarCheck').checked = params.useHomeKeyToFocusSearchBar;
+document.getElementById('openExternalLinksInNewTabsCheck').checked = params.openExternalLinksInNewTabs;
+switchHomeKeyToFocusSearchBar();
+document.getElementById('bypassAppCacheCheck').checked = !params.appCache;
+document.getElementById('appVersion').textContent = 'Kiwix ' + params.appVersion;
+// We check here if we have to warn the user that we switched to ServiceWorkerMode
+// This is only needed if the ServiceWorker mode is available, or we are in an Extension that supports Service Workers
+// outside of the extension environment, AND the user's settings are stuck on jQuery mode, AND the user has not already been
+// alerted about the switch to ServiceWorker mode by default
+if ((isServiceWorkerAvailable() || isMessageChannelAvailable() && /^(moz|chrome)-extension:/i.test(window.location.protocol)) &&
+    params.contentInjectionMode === 'jquery' && !params.defaultModeChangeAlertDisplayed) {
+    // Attempt to upgrade user to ServiceWorker mode
+    params.contentInjectionMode = 'serviceworker';
+} else if (params.contentInjectionMode === 'serviceworker') {
+    // User is already in SW mode, so we will never need to display the upgrade alert
+    params.defaultModeChangeAlertDisplayed = true;
+    setSetting('defaultModeChangeAlertDisplayed', true);
+}
+if (!/^chrome-extension:/i.test(window.location.protocol)) {
+    document.getElementById('serviceWorkerLocal').style.display = 'none';
+    document.getElementById('serviceWorkerLocalDescription').style.display = 'none';
+}
+setContentInjectionMode(params.contentInjectionMode);
```

Comment:
Regarding your query, just move this line back to app.js. You copied a bit too much in this case, it doesn't need to be in init.js.